Source/WebCore/inspector/InspectorStyleSheet.cpp

```diff
@@ -1112,7 +1122,7 @@ ExceptionOr<CSSStyleRule*> InspectorStyleSheet::addRule(const String& selector)
     if (!m_pageStyleSheet)
         return Exception { NotSupportedError };
 
-    if (!isValidRuleHeaderText(selector, StyleRuleType::Style, m_pageStyleSheet->ownerDocument()))
+    if (!isValidRuleHeaderText(selector, StyleRuleType::Style, m_pageStyleSheet->ownerDocument(), CSSSelectorParser::IsNestedContext::No))
```

Comment:
> When used in the selector of a [nested style rule](https://drafts.csswg.org/css-nesting/#nested-style-rule), the [nesting selector](https://drafts.csswg.org/css-nesting/#nesting-selector) represents the elements matched by the parent rule. When used in any other context, it represents nothing. (That is, it’s valid, but matches no elements.)

Actually the nesting selector & is allowed everywhere (even inside top-level rules), so maybe we don't need this parameter.

Reply:
But the relative selector syntax is not allowed everywhere though, so maybe we do need this here (depending of the goal of the inspector)

Reply:
On the line you commented on here, we will always be creating a new top-level style rule, so the relative syntax wouldn’t be allowed if I understand correctly. Above where we allow editing the selector, we dynamically determine if nesting-only syntax should be allowed. Our goal is to ensure that an edit to a selector maintains the validity of the overall style sheet (even if the rule no longer applies to the element after the edit). We don’t want developers to be able to make an edit they can’t fix later because the style is no longer syntactically valid, preventing us from parsing it and potentially other styles.